### ISSUE TITLE
HCP doesn't work after upgrade Meteor 1.4.2.3 to 1.7b26

### ISSUE THREAD

**Issue by crapthings** [meteor/meteor#9846]
i have tried this one

https://github.com/meteor/meteor/commit/a599aa738d78e22034a91ca93592cea2a722e393

https://github.com/meteor/meteor/issues/9574

**Comment by crapthings** [comment]
![image](https://user-images.githubusercontent.com/1147704/39390725-722dc6f8-4acb-11e8-9d05-02963ce85ced.png)

```
Error: Error downloading asset manifest: Error Domain=NSURLErrorDomain Code=-1001 "The request timed out." UserInfo={NSUnderlyingError=0x170445340 {Error Domain=kCFErrorDomainCFNetwork Code=-1001 "(null)" UserInfo={_kCFStreamErrorCodeKey=-2102, _kCFStreamErrorDomainKey=4}}, NSErrorFailingURLStringKey=http://192.168.1.220:3000/__cordova/manifest.json, NSErrorFailingURLKey=http://192.168.1.220:3000/__cordova/manifest.json, _kCFStreamErrorDomainKey=4, _kCFStreamErrorCodeKey=-2102, NSLocalizedDescription=The request timed out.}
```

**Comment by crapthings** [comment]
meteor 171

"cordovaCompatibilityVersions": {
"android": "0dc4f1a89227c366096967faa0ba1db5ef253ae7",
"ios": "d0a9b84b258e7382bb54eaaa670d9bb59040a3ce"
}

meteor 142
"cordovaCompatibilityVersions": {
"android": "98fe946c669db663ce92105a4e58275273a76667",
"ios": "004403d8e809d48a90ee8d88bbfbe240abe8da51"
}

**Comment by crapthings** [comment]
Download failure: Non-success status code 404 for asset: /__cordova/packages/mongo-id.js

everytime we try to download new bundle, the 404 is different.

and its trying to download assets from android folder ?

![image](https://user-images.githubusercontent.com/1147704/39392192-4b3de816-4ae4-11e8-972a-3bed9b38b12e.png)

**Comment by crapthings** [comment]
i can confirm HCP works with this commit

https://github.com/crapthings/meteor/commit/498f5d32dd72d0cd480bcfe3e7df98b6eb3441be

METEOR_CORDOVA_COMPAT_VERSION_IOS=004403d8e809d48a90ee8d88bbfbe240abe8da51 /Users/monsterstep/dev/sandbox-experimentals/workmeteor/meteor/./meteor run --mobile-server http://10.0.0.7:3000

**Comment by crapthings** [comment]
i can confirm that HCP work with this commit

https://github.com/crapthings/meteor/commit/498f5d32dd72d0cd480bcfe3e7df98b6eb3441be

### the problem is ROOT URL and MOBILE-SERVER

ROOT_URL is more dynamic than MOBILE_SERVER because it can be set with env ROOT_URL, then it will write ROOT_URL to HTML HEAD

![image](https://user-images.githubusercontent.com/1147704/39394080-1252c424-4b00-11e8-8c27-9e96bfda5254.png)

and we can put a package on top of all package to replace the ROOT_URL to achieve switch between server on cordova app

![image](https://user-images.githubusercontent.com/1147704/39394116-74dd50be-4b00-11e8-9e27-c61979985812.png)

https://github.com/crapthings/meteor-switch-root-url/blob/master/index.js

BUT MOBILE-SERVER is not, it will build into

https://github.com/meteor/cordova-plugin-meteor-webapp/blob/8bbf1aefa64f424896b3fc0aaf54775f313177b5/src/ios/AssetBundleManager.swift#L95,

which cause HCP only work with that URL, so if we switch to new server on cordova app, the MOBILE-SERVER is still the build one, that leads HCP failed.

![image](https://user-images.githubusercontent.com/1147704/39394142-c10cdd88-4b00-11e8-9e07-69c5c86b5cdb.png)

**Comment by crapthings** [comment]
after dig around i've found

![image](https://user-images.githubusercontent.com/1147704/39394348-d5ff32ec-4b03-11e8-9239-77eefaaa54d9.png)

**Comment by crapthings** [comment]
@benjamn 

METEOR@1.7-beta.28 works~

**Comment by fizx1** [comment]
@benjamn 

Hi Ben! Thanks for your work on this issue! Sorry for my ignorance about the Meteor version numbering system, but is this issue fixed in the latest release version, number 1.7.0.5? Thanks!!!
